vita3k/ctrl/include/ctrl/state.h

```diff
@@ -35,10 +35,11 @@ typedef std::shared_ptr<_SDL_Haptic> HapticPtr;
 
 struct Controller {
     GameControllerPtr controller;
-    int port;
+    int port; // 1-based
```

Comment:
Are you sure making it 1-base is the best? for allmost all uses you need to add/remove 1 to it.

Reply:
That part is legacy, and I left only the comments to prevent confusion for future developers after experiencing confusion myself with port numbers switching between 1-based and 0-based during development. The original 1-based port numbering seems to have been done to match it with sceCtrl.
However, as you mentioned, there's a lot of code getting complicated just to maintain that part. I think refactoring all parts including the existing code to be 0-based would help prevent future mistakes. I'll proceed with additional development to address this as well.